audio-worklet/test/bitcrusher-test.js

```diff
@@ -0,0 +1,129 @@
+let a = [0, 1, 2, 3, 4];
+let b = [5, 6, 7, 8, 9];
+let bitcrusher = new Bitcrusher(context);
+let outputA = [];
+let outputB = [];
+
+let factor = 4;
+bitcrusher.processBuffer_(factor, 24, a, outputA);
+bitcrusher.processBuffer_(factor, 24, b, outputB);
+
+let output = outputA.concat(outputB);
+
+output.map((data, index) => {
+  let expected = Math.floor(index / factor) * factor;
+  console.assert(
+      data === expected, 'computed ' + data + ' but expected ' + expected);
+});
+
+// Create one oscillator, and connect it to a bitcrusher with variables set
+// to avoid any effect and a delay node which compensates for the script
+// processor latency. Then verify that the samples are nearly identical.
+reduction = 1;
+bitDepth = 24;
+let bufferLength = 512;
+let oscillator = new OscillatorNode(context);
+oscillator.start();
+
+let merger = new ChannelMergerNode(context, {numberOfInputs: 2});
+bitcrusher = new Bitcrusher(context, {
+  buffersize: bufferLength,
+  inputChannels: 1,
+  outputChannels: 1,
+  bitDepth: bitDepth,
+  reduction: reduction
+});
+
+// Accomodate for script processor latency by delaying the unaltered oscillator.
+let delay = context.createDelay();
+delay.delayTime.value = bufferLength / context.sampleRate;
+
+oscillator.connect(bitcrusher.input);
+bitcrusher.output.connect(merger, 0, 0);
+oscillator.connect(delay).connect(merger, 0, 1);
+merger.connect(context.destination);
+
+// When audio buffer is ready, verify bitcrushed samples are unaltered.
+context.startRendering()
+    .then((buffer) => {
+      let bitcrusherOutput = buffer.getChannelData(0);
+      let unalteredOutput = buffer.getChannelData(1);
+
+      // Allow for fractional error beyond audible perception. This error
+      // occurs because any sample passing through bitcrusher will undergo
+      // manipulation in Math.round() and will therefore be represented by a
+      // new floating point number that differs slightly from the original.
+      // In tested  samples, the maximum observed error is 3.46451997756958e-7
+      const permittedSampleError = 3.46451997756958e-7 + 1e-7;
+
+      // Verify samples from unadultered oscillator match samples from
+      // bitcrushed oscillator with non-information reducing parameters.
+      for (let i = 0; i < unalteredOutput.length; i++) {
+        let crushedSample = bitcrusherOutput[i];
+        let unalteredSample = unalteredOutput[i];
+        const diff = Math.abs(unalteredSample - crushedSample);
+
+        if (i < bufferLength) {
+          console.assert(
+              bitcrusherOutput[i] == 0,
+              'Bitcrusher sample at ' + i + 'expected to be 0 but it was ' +
+                  bitcrusherOutput[i]);
+        }
+
+        console.assert(
+            diff < permittedSampleError,
+            'Bitcrushed sample at ' + i + ' is ' + crushedSample + ' but ' +
+                unalteredSample + ' for oscillator');
+      }
+    })
+    .catch((err) => {
```

Comment:
Not a big deal, but I don't think `catch` is necessary. This is a test, so if something goes wrong it would throw in the console anyway.